### ISSUE TITLE
umount leaves a directory in the vm

### ISSUE THREAD

**Issue by jasonmccallister** [canonical/multipass#1527]
**Describe the bug**
When removing a mount the directory is left in the virtual machine

**To Reproduce**
1. `multipass mount /Users/jasonmccallister/dev/issue96 issue96:/nitro/dev`
1. Remove the mount and see the dir `/nitro/dev` is still there

**Expected behavior**
The `/nitro/dev` directory to not exist.

**Additional info**
 - OS: macOS 10.15.4
- `multipass version 1.2.1`

```
Name:           issue96
State:          Running
IPv4:           192.168.64.63
Release:        Ubuntu 18.04.4 LTS
Image hash:     3b2e3aaaebf2 (Ubuntu 18.04 LTS)
Load:           0.14 0.24 0.15
Disk usage:     2.4G out of 38.6G
Memory usage:   352.2M out of 3.9G

Name:           nitro-dev
State:          Running
IPv4:           192.168.64.21
Release:        Ubuntu 18.04.4 LTS
Image hash:     2f6bc5e7d9ac (Ubuntu 18.04 LTS)
Load:           0.06 0.06 0.01
Disk usage:     2.3G out of 38.6G
Memory usage:   222.7M out of 3.9G
Mounts:         /Users/jasonmccallister/dev/craft-dev                              => /nitro/sites/nitro-demo.test
                    UID map: 501:default
                    GID map: 20:default
                /Users/jasonmccallister/go/src/github.com/craftcms/nitro/demo-site => /nitro/sites/demo-site
                    UID map: 501:default
                    GID map: 20:default
```

**Comment by Saviq** [comment]
Hi @jasonmccallister, the problem is there's loads of things that could happen between creating and removing a mount. We'd have to remember the exact path we had to `mkdir` and then remove up to that point on destroy.

But what if you mounted partially overlapping paths and removing it in reverse order? It will be busy, so we can't remove the common path, and you'll be left with that on the unmount…

IMO it feels like a lot of complexity for very little gain.

**Comment by jasonmccallister** [comment]
I agree bunch of things can go wrong. In our use case - if someone tries to mount the directory again - it throws an error. We can certainly remove this directory in our project, but someone else might encounter this error.

Maybe an optional flag on the `umount` command to clean up the directory?

**Comment by Saviq** [comment]
Hmm can you explain what you mean by an error being thrown when mounting again?
